paperqa/settings.py

```diff
@@ -58,8 +58,7 @@
 from paperqa.utils import hexdigest, pqa_directory
 from paperqa.version import __version__
 
-if TYPE_CHECKING:
-    from .agents.env import EnvironmentState
+EnvironmentState = TypeVar("EnvironmentState")
```

Comment:
Why does this fix work? It seems like we'd need to make `AgentSettings` a `Generic[EnvironmentState]`

Reply:
importing EnvironmentState introduces a circular dependency which breaks things.
Unless I copy that class into `settings.py`, I don't believe `Generic[EnvironmentState]` will work.

When I added the type_checking originally, that's what I was aiming to resolve.

Reply:
Yeah we need to bite the bullet and resolve the circular dependency. I don't think this current fix here involving a `TypeVar` is actually valid.

I think we may need to move `EnvironmentState` + `make_status` to be in `settings.py`, and import them into `tools.py`

Reply:
gotcha. on it!